File: 2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        l3 = ListNode(0)
        head = l3
        tmp = 0
        while l1 != None or l2 != None or tmp != 0:
            if l1 != None:
                tmp += l1.val
                l1 = l1.next
            if l2 != None:
                tmp += l2.val
                l2 = l2.next
            l3.next =  ListNode(tmp % 10)
            l3 = l3.next
            tmp = tmp // 10
        return head.next;

        
        

File: 2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        int tmp1 = 0, tmp2 = 0, x = 1;
        ListNode *l3 = NULL, *head = NULL;
/*        while(l1 != NULL){
            tmp1 += (l1->val) * x;
            x *= 10;
            l1 = l1 -> next;
        }
        x = 1;
        while(l2 != NULL){
            tmp2 += (l2->val) * x;
            x *= 10;
            l2 = l2 -> next;
        }
        int tmp = tmp1 + tmp2, flag = 1;
*/
        while(l1||l2||tmp){
            if(l1){
                tmp += l1 -> val;
                l1 = l1 -> next;
            }
            if(l2){
                tmp += l2 -> val;
                l2 = l2 -> next;
            }
            if(l3) {
                l3->next = new ListNode(tmp%10);
                l3 = l3->next;
            }
            else {
                l3 = head = new ListNode(tmp%10);
            }
            tmp /= 10;
        }
        //printf("%d %d", tmp1, tmp2);
        return head;
    }
};